### PR TITLE
Implemented inventory expansion

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: #5218

* **Server Mode**: Both

* **Description of Pull Request**: 
Implements the inventory expansion feature.

Credits to @4144 and @Asheraf for their packets provided in packet_struct.hpp

### CODE REVIEW

File: src/map/itemdb.hpp [hunk 1]
```diff
@@ -122,6 +122,7 @@ enum item_itemid : t_itemid
 	ITEMID_WOB_RACHEL					= 14584,
 	ITEMID_WOB_LOCAL					= 14585,
 	ITEMID_SIEGE_TELEPORT_SCROLL		= 14591,
+	ITEMID_INVENTORY_EXPANSION_COUPON	= 25793,
```
Comment (RagnaWay) [inline src/map/itemdb.hpp:125]: add too 25791 ( https://www.divine-pride.net/database/item/25791 )

File: src/map/pc.hpp [hunk 2]
```diff
@@ -1049,9 +1051,15 @@ extern JobDatabase job_db;
...
 #define pc_istrading(sd)      ( (sd)->npc_id || (sd)->state.vending || (sd)->state.buyingstore || (sd)->state.trading )
+static bool pc_cant_act2( struct map_session_data* sd ){
+	return sd->state.vending || sd->state.buyingstore || (sd->sc.opt1 && sd->sc.opt1 != OPT1_BURNING)
+		|| sd->state.trading || sd->state.storage_flag || sd->state.prevend || sd->state.refineui_open
+		|| sd->state.inventory_expansion_confirmation;
+}
 // equals pc_cant_act2 and additionally checks for chat rooms and npcs
-#define pc_cant_act(sd)       ( (sd)->npc_id || (sd)->chatID || pc_cant_act2( (sd) ) )
-#define pc_cant_act2(sd)      ( (sd)->state.vending || (sd)->state.buyingstore || ((sd)->sc.opt1 && (sd)->sc.opt1 != OPT1_BURNING) || (sd)->state.trading || (sd)->state.storage_flag || (sd)->state.prevend || (sd)->state.refineui_open )
+static bool pc_cant_act( struct map_session_data* sd ){
```
[the remark this answers is not in the document]
Comment (Atemo) [reply src/map/pc.hpp:1060]: What do you mean? your comment is not clear

[the remark this answers is not in the document]
Comment (RagnaWay) [reply src/map/pc.hpp:1060]: This same addition has the same in StylistUI
image

**Comment by Seavoltos** [comment]
When I connect, it does not update the maximum number of objects, I can modify it correctly using coupons, but they are not saved, what could it be
Screenshot_1
?

File: src/map/pc.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Atemo) [reply src/map/pc.hpp:1060]: The 2 PR are independent there is no issue here

[the remark this answers is not in the document]
Comment (Lemongrass3110) [reply src/map/pc.hpp:1060]: This is intended to prevent bigger conflicts.

**Comment by Lemongrass3110** [comment]
> When I connect, it does not update the maximum number of objects, I can modify it correctly using coupons, but they are not saved, what could it be Screenshot_1 ?

@Seavoltos you merged incorrectly. Check your char server and see if you have error messages there.

File: src/map/itemdb.hpp (hunk 1 above)
Comment (Lemongrass3110) [reply src/map/itemdb.hpp]: Done. Thx.

File: src/map/pc.cpp [hunk 3]
```diff
@@ -5103,11 +5108,12 @@ char pc_checkadditem(struct map_session_data *sd, t_itemid nameid, int amount)
  *------------------------------------------*/
 uint8 pc_inventoryblank(struct map_session_data *sd)
 {
-	uint8 i, b;
+	uint16 i;
+	uint8 b;
```
Comment (aleos89) [inline src/map/pc.cpp:5112]:
```suggestion
```

File: src/map/pc.cpp [hunk 4]
```diff
@@ -5103,11 +5108,12 @@ char pc_checkadditem(struct map_session_data *sd, t_itemid nameid, int amount)
...
 
 	nullpo_ret(sd);
 
-	for(i = 0, b = 0; i < MAX_INVENTORY; i++){
+	for(i = 0, b = 0; i < sd->status.inventory_slots; i++){
```
Comment (aleos89) [inline src/map/pc.cpp:5116]:
```suggestion
	uint8 b;

	for(uint16 i = 0, b = 0; i < sd->status.inventory_slots; i++){
```

**Review by aleos89 — APPROVED** [review]
Minor nit picks, the rest looks good!

File: src/map/pc.cpp (hunk 4 above)
Comment (aleos89) [reply src/map/pc.cpp:5116]: Ignore this, totally missed `, b = 0` in there.

**Comment by gravity-ro** [comment]
> > When I connect, it does not update the maximum number of objects, I can modify it correctly using coupons, but they are not saved, what could it be Screenshot_1 ?
> 
> @Seavoltos you merged incorrectly. Check your char server and see if you have error messages there.

I can't solve
everything fine no error but inventory it's still 0 
how you solved it.
Thank you advance.

**Comment by Balferian** [comment]
@gravity-ro do you update your `char` table in db?

**Comment by gravity-ro** [comment]
> @gravity-ro do you update your `char` table in db?

yes i do and i use item expansion it's not remember when login again.

**Comment by Balferian** [comment]
@gravity-ro write me in discord please (Balfear#3330) or open new issue

**Comment by gravity-ro** [comment]
> @gravity-ro write me in discord please (Balfear#3330) or open new issue

Thank you i start a new issue.

**Comment by Lemongrass3110** [comment]
This is no support system. Please use Discord or the board.

**Pull request merged**